compiler/rustc_lint/src/builtin.rs

```diff
@@ -1635,6 +1627,51 @@ impl<'tcx> LateLintPass<'tcx> for TrivialConstraints {
     }
 }
 
+declare_lint! {
+    /// The `double_negation` lint detects expressions of the form `--x`.
+    ///
+    /// ### Example
+    ///
+    /// ```rust
+    /// fn main() {
+    ///     let x = 1;
+    ///     let _b = --x;
+    /// }
+    /// ```
+    ///
+    /// {{produces}}
+    ///
+    /// ### Explanation
+    ///
+    /// Negating something twice is usually the same as not negating it at all.
+    /// However, a double negation in Rust can easily be confused with the
+    /// prefix decrement operator that exists in many languages derived from C.
+    /// Use `-(-x)` if you really wanted to negate the value twice.
+    ///
+    /// To decrement a value, use `x -= 1` instead.
+    pub DOUBLE_NEGATION,
```

Comment:
Better name, as by the lint naming conventions:
```suggestion
    pub DOUBLE_NEGATIONS,
```

Reply:
You mean [this](https://rust-lang.github.io/rfcs/0344-conventions-galore.html#lints)? Because `#[allow(double_negation)]` seems grammatically fine to me. But I'll change it anyway, it does say that you should use plurals for lint names